sdk/go/pulumi/resource.go

```diff
@@ -217,6 +217,17 @@ func (o resourceOrInvokeOption) applyInvokeOption(opts *invokeOptions) {
 	o(nil, opts)
 }
 
+// merging is handled by each functional options call
+// properties that are arrays/maps are always appened/merged together
+// last value wins for non-array/map values and for conflicting map values (bool, struct, etc)
+func merge(opts ...ResourceOption) *resourceOptions {
```

Comment:
I believe a desired part of this was to expose a public API equivalent to `mergeOptions` in NodeJS. This is frequently used by user code on (a) transformations (b) some component scenarios that need to mix in options to what the user provided. The existing tests for (a) could/should probably be updated to use this API. Related - is https://www.pulumi.com/docs/intro/concepts/programming-model/#transformations “correct” wrt these changes?  If so, perhaps you his merging doesn’t need to be exposed to users?

Reply:
Yes, the transformations example is still correct. With the way we implemented functional options, we don't expose the underlying `resourceOptions` struct. This implementation more or less enables equivalent behavior with nodejs. The only difference is that you have no way to introspect the existing options, delete options, etc. 

If we're fine with that caveat, variadic args allows the omission of an explicit `mergeOptions` API.